src/Functions/UserDefined/UserDefinedDriverFunctionFactory.cpp

```diff
@@ -0,0 +1,290 @@
+
+    if (UserDefinedSQLFunctionFactory::instance().has(function_name))
+        throw Exception(ErrorCodes::FUNCTION_ALREADY_EXISTS, "User defined SQL function '{}' already exists", function_name);
+
+    if (UserDefinedExecutableFunctionFactory::instance().has(function_name, global_context)) /// NOLINT(readability-static-accessed-through-instance)
+        throw Exception(ErrorCodes::FUNCTION_ALREADY_EXISTS, "User defined executable function '{}' already exists", function_name);
+
+    checkDriverExists(query);
+}
+
+void UserDefinedDriverFunctionFactory::checkCanBeUnregistered(const String & function_name)
+{
+    if (FunctionFactory::instance().hasNameOrAlias(function_name) || AggregateFunctionFactory::instance().hasNameOrAlias(function_name))
+        throw Exception(ErrorCodes::CANNOT_DROP_FUNCTION, "Cannot drop system function '{}'", function_name);
+
+    if (UserDefinedSQLFunctionFactory::instance().has(function_name))
+        throw Exception(ErrorCodes::CANNOT_DROP_FUNCTION, "Cannot drop user defined SQL function '{}'", function_name);
+
+    if (UserDefinedExecutableFunctionFactory::instance().has(function_name, global_context)) /// NOLINT(readability-static-accessed-through-instance)
+        throw Exception(ErrorCodes::CANNOT_DROP_FUNCTION, "Cannot drop user defined executable function '{}'", function_name);
+}
+
+void UserDefinedDriverFunctionFactory::checkDriverExists(const ASTPtr & query) const
+{
+    auto create_driver_function = query->as<ASTCreateDriverFunctionQuery &>();
+    auto engine_name = create_driver_function.getEngineName();
+
+    auto names = UserDefinedDriverFactory::instance().getRegisteredNames(global_context); /// NOLINT(readability-static-accessed-through-instance)
+
+    if (!UserDefinedDriverFactory::instance().has(engine_name, global_context)) /// NOLINT(readability-static-accessed-through-instance)
+        throw Exception(ErrorCodes::UNSUPPORTED_DRIVER, "Cannot find a driver with engine name '{}', drivers size = {}", engine_name, names.size());
+}
+
+UserDefinedExecutableFunctionPtr UserDefinedDriverFunctionFactory::createUserDefinedFunction(
+    const ASTCreateDriverFunctionQuery & query, const UserDefinedDriverPtr & driver) const
+{
+    const auto & configuration = driver->getConfiguration();
+
+    std::vector<UserDefinedExecutableFunctionArgument> arguments;
+    if (query.function_params)
+    {
+        for (auto & child : query.function_params->children)
+        {
+            auto * column = child->as<ASTNameTypePair>();
+            arguments.emplace_back(DataTypeFactory::instance().get(column->type), column->name);
+        }
+    }
+
+    auto result_type = DataTypeFactory::instance().get(query.function_return_type->as<ASTDataType>()->name);
+
+    auto command = configuration.command;
+    auto command_arguments = configuration.command_arguments;
+
+    const auto & code = query.getFunctionBody();
+    if (command_arguments.empty())
+    {
+        command += " \"" + code + '\"';
```

Comment:
This won't work for two reasons:

1. A limit on the command length.
2. Different shells may require different escaping.

This needs a solution with temporary files.